shared-module/vectorio/Polygon.c

```diff
@@ -12,18 +14,65 @@
 // #define VECTORIO_POLYGON_DEBUG(...) mp_printf(&mp_plat_print __VA_OPT__(,) __VA_ARGS__)
 
 
+// Converts a list of points tuples to a flat list of ints for speedier internal use.
+// Also validates the points.
+static void _clobber_points_list(vectorio_polygon_t *self, mp_obj_t points_tuple_list) {
+    size_t len = 0;
+    mp_obj_t *items;
+    mp_obj_list_get(points_tuple_list, &len, &items);
+    VECTORIO_POLYGON_DEBUG("polygon_points_list len: %d\n", len);
+
+    if ( len < 3 ) {
+        mp_raise_TypeError_varg(translate("Polygon needs at least 3 points"));
+    }
+
+    if ( self->len < 2*len ) {
+        if ( self->points_list != NULL ) {
+            gc_free( self->points_list );
+        }
+        self->points_list = gc_alloc( 2 * len * sizeof(int), false, false );
+    }
+    self->len = 2*len;
+
+    for ( size_t i = 0; i < len; ++i) {
+        size_t tuple_len = 0;
+        mp_obj_t *tuple_items;
+        mp_obj_tuple_get(items[i], &tuple_len, &tuple_items);
+
+        if (tuple_len != 2) {
+            mp_raise_ValueError_varg(translate("%q must be a tuple of length 2"), MP_QSTR_point);
+        }
+        if (!mp_obj_get_int_maybe(tuple_items[0], &self->points_list[2*i])) {
+            self->len = 0;
+            gc_free( self->points_list );
+            self->points_list = NULL;
+            mp_raise_ValueError_varg(translate("unsupported %q type"), MP_QSTR_point);
+        }
+        if (!mp_obj_get_int_maybe(tuple_items[1], &self->points_list[2*i + 1])) {
+            self->len = 0;
+            gc_free( self->points_list );
+            self->points_list = NULL;
+            mp_raise_ValueError_varg(translate("unsupported %q type"), MP_QSTR_point);
+        }
```

Comment:
```suggestion
        if (!mp_obj_get_int_maybe(tuple_items[0], &self->points_list[2*i]) ||
            !mp_obj_get_int_maybe(tuple_items[1], &self->points_list[2*i + 1])) {
            self->len = 0;
            gc_free( self->points_list );
            self->points_list = NULL;
            mp_raise_ValueError_varg(translate("unsupported %q type"), MP_QSTR_point);
        }
```

Reply:
Still looks like it has the duplication. Did you accept the suggestion without pulling it down to your local copy?

Reply:
Yes I did accept the suggestion without pulling it down to my local copy.  🤦